toolchain/sem_ir/resolve_node_id.h

```diff
@@ -0,0 +1,33 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_SEM_IR_RESOLVE_NODE_ID_H_
+#define CARBON_TOOLCHAIN_SEM_IR_RESOLVE_NODE_ID_H_
+
+#include "toolchain/parse/tree_and_subtrees.h"
+#include "toolchain/sem_ir/file.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::SemIR {
+
+// A specific node location in a file.
+struct ResolvedNodeId {
+  CheckIRId check_ir_id;
+  Parse::NodeId node_id;
+};
```

Comment:
Hmm... We're starting to use the term "resolve" in Check/SemIR in other ways, and so I'm nervous about calling these `ResolvedNodeId`s... @zygoloid do you have ideas for a better naming approach here? (Or in the other places we're starting to use "resolve"?)

Reply:
I renamed to "CanonicalNodeId", is that better? I think most words we'd choose in this area are going to be used elsewhere in other ways -- canonical is also used for ImportIRInst.

Reply:
Adding @zygoloid as a reviewer, too.

Reply:
I think "canonical" is OK. I suggested "absolute" in person too (by analogy to an absolute path), but either works for me.

Reply:
I think "canonical" implies a bit to much that there is a canonicalization process, and there might be non-canonical ones...

I like "absolute", and particularly like the analogy of an absolute path to help explain.

Reply:
Changed to absolute since you both agree on it, but the path analogy doesn't work for me.

Note, the absolute path analogy doesn't work for me. If we have a symlink `/bin/carbon -> /usr/lib/carbon/carbon-busybox`, `/bin/carbon` is an "absolute path" because it starts with a "/". However, the "real path" is `/usr/lib/carbon/carbon-busybox`. `std::filesystem::canonical` handles symlinks, `std::filesystem::absolute` would just turn "carbon" into "/bin/carbon"

For me, `LocId` is closer to a symlink that we're asking to canonicalize.